ueberdosis/tiptap
# feat(suggestion): fallback to extracting previous node text content if it not a text node

## Please describe your changes

Change `findSuggestionMatch` to extract text from the previous node, even if the node's type is not text. The original implementation only extracts text from the previous node if it is a node of type text. This means that if the previous node is a node that wraps a text node, the text will not be extracted, and thus the suggestion list cannot launch. The change here is backwards compatible, and just adds a fallback for text extraction for this extra case, making the suggestion plugin a little more flexible.

## How did you accomplish your changes

Changed `findSuggestionMatch` to use `textContent` property to extract text from the previous node when that node is not a text node. 

## How have you tested your changes

Manually tested the plugin with a wrapped text node, and it launches the plugin as expected. I raised this issue because our team implemented a hashtag node with continuous recognition, with a structure like this:
```
{
  type: 'hashtag'
  content: [
    {
      type: 'text'
      hashtag: '#hashtagtext'
    }
  ]
}
```
The continuous recognition meant adding a plugin to listen to `handleTextInput`, and transform the preceding text into a hashtag node if it matched the regex. However, doing this means that as you type, the previous text can become wrapped in a node, which still should be scanned by the suggestion plugin for text to match against. 

## How can we verify your changes

Add a node to type that wraps text, and insert it to the editor, making sure that the text begins with the suggestion plugin character. Type another character after this node; the suggestion list should open.

## Checklist


## Related issues

https://github.com/ueberdosis/tiptap/issues/4779
